sdv/constraints/base.py

```diff
@@ -198,9 +182,22 @@ def transform(self, table_data):
             pandas.DataFrame:
                 Input data unmodified.
         """
-        self._validate_data_on_constraint(table_data)
-        self.check_missing_columns(table_data)
-        return self._transform(table_data)
+        self._use_reject_sampling = False
+        self._validate_all_columns_present(table_data)
+
+        try:
+            transformed = self._transform(table_data)
+            if self.is_custom:
+                self.reverse_transform(transformed)
+            return transformed
+
+        except Exception:
```

Comment:
Can you use the specific errors?

Reply:
we want to catch all possible exceptions here